src/main/java/org/kiwiproject/champagne/resource/AuthResource.java

```diff
@@ -0,0 +1,63 @@
+package org.kiwiproject.champagne.resource;
+
+import com.codahale.metrics.annotation.ExceptionMetered;
+import com.codahale.metrics.annotation.Timed;
+import lombok.AllArgsConstructor;
+import org.dhatim.dropwizard.jwt.cookie.authentication.DefaultJwtCookiePrincipal;
+import org.dhatim.dropwizard.jwt.cookie.authentication.JwtCookiePrincipal;
+import org.kiwiproject.champagne.jdbi.UserDao;
+import org.kiwiproject.jaxrs.exception.JaxrsNotAuthorizedException;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+/**
+ * Resource for various authentication based endpoints
+ */
+@Path("/auth")
+@AllArgsConstructor
+@Consumes(MediaType.APPLICATION_JSON)
+@Produces(MediaType.APPLICATION_JSON)
+public class AuthResource {
+
+    private final UserDao userDao;
+
+    @POST
+    @Path("/login")
+    @Timed
+    @ExceptionMetered
+    public DefaultJwtCookiePrincipal login(@Context ContainerRequestContext context, String name) {
+
+        // TODO: Need to actually check password or valid login not just user lookup
+        var user = userDao.findBySystemIdentifier(name);
+
+        // TODO: Since we are setting the context, we should talk about whether we want the response to be
+        //       the actual principal since that essentially contains the JWT information.
+        return user.map(u -> {
+            // By calling addInContext, the principal is set on the SecurityContext and a cookie is generated
+            var principal = new DefaultJwtCookiePrincipal(u.getSystemIdentifier());
+            principal.addInContext(context);
+            return principal;
+        }).orElseThrow(() -> new JaxrsNotAuthorizedException("Invalid login"));
+    }
+
+    @DELETE
+    @Path("/logout")
+    @Timed
+    @ExceptionMetered
+    public Response logout(@Context ContainerRequestContext context) {
+        // From the dropwizard-jwt-cookie-authentication docs:
+        // It is a stateless authentication method, so there is no real way to invalidate a session other than
+        // waiting for the JWT to expire. However, calling JwtCookiePrincipal.removeFromContext(context) will
+        // make browsers discard the cookie by setting the cookie expiration to a past date.
```

Comment:
```suggestion
        // make browsers discard the cookie by setting the cookie expiration to a previous date.
```

Maybe the comments re: `addInContext` and `removeFromContext` should be removed? Otherwise I can see them being repeated in each service...

Reply:
I can remove the comments, I had put them there so that you (@sleberknight) didn't have to go view the library documentation for this review.

Reply:
Ha ok. In this case you can create the PR and then add comments on your own PR (which I have been known to do on occasion). So yes we should remove the comments explaining the library.